control/state.py

```diff
@@ -222,6 +226,55 @@ def get_omap_version(self) -> int:
                 f" invalid number of values ({value_list}).")
             raise
 
+    def lock_omap(self, timeout = None):
+        got_lock = False
+        need_to_unlock = False
```

Comment:
I'm uncertain about variable `need_to_unlock`. I notice it's initially set to False here and then switched to true on line 242 bellow. Nonetheless, I can't find any part of the code that actually reads its value.